src/sentry/management/commands/makemigrations.py

```diff
@@ -15,14 +16,34 @@
 """
 
 
+# We check that the latest migration is the one stored in the lockfile
+def validate(migrations_filepath, latest_migration_by_app):
+    infile = {}
+    with open(migrations_filepath, encoding="utf-8") as file:
+        for line in file:
+            try:
+                app_label, name = line.split(": ")
+                infile[app_label] = name.strip()
+            except ValueError:
+                pass
+
+    for app_label, name in sorted(latest_migration_by_app.items()):
+        if infile[app_label] != name:
+            print(  # noqa: B314
+                f"ERROR: The latest migration does not match the one in the lockfile -> `{app_label}` app: {name} vs {infile[app_label]}"
+            )
+            # makemigrations.Command --check exits with 1 if a migration needs to be generated
+            sys.exit(2)
+
+
 class Command(makemigrations.Command):
     """
     Generates a lockfile so that Git will detect merge conflicts if there's a migration
     on master that doesn't exist in a branch.
     """
 
     def handle(self, *app_labels, **options):
-        if not options["name"]:
+        if not options["name"] and not options.get("check_changes"):
```

Comment:
Is the option `check_changes` or `check`?

Reply:
Surprisingly `check_changes`.